### PR TITLE
repair and speed up CRAN version tests

### PR DESCRIPTION
**Pull request by TSchiefer**
- repaired tests in one commit
- tried removing redundant tests
- tried removing redundant setup code

test now consistently under 1min; 
one slow-ish test will be additionally skipped_on_cran.

### CODE REVIEW

File: tests/testthat/helper-expectations.R [hunk 1]
```diff
@@ -18,9 +18,9 @@ expect_dm_error <- function(expr, class) {
 }
 
 expect_name_repair_message <- function(expr) {
-  # FIXME: Name repair should get a message
+  # Name repair did not get a message during some time in {vctrs}
   # https://github.com/r-lib/vctrs/issues/849
-  if (packageVersion("vctrs") >= "0.2.3") {
+  if (packageVersion("vctrs") >= "0.2.3" && packageVersion("vctrs") < "0.2.99.9006") {
```
Comment (krlmlr) [inline tests/testthat/helper-expectations.R]:
```suggestion
  if (packageVersion("vctrs") < "0.2.99.9006") {
```

File: tests/testthat/test-db-interface.R [hunk 2]
```diff
@@ -19,11 +19,15 @@ test_that("copy_dm_to() copies data frames from databases", {
 })
 
 test_that("copy_dm_to() copies between sources", {
+  # speed things up on cran
+  skip_on_cran()
+
+  all_combos <- crossing(tibble(a = 1:3), tibble(b = 1:3))
```
Comment (krlmlr) [inline tests/testthat/test-db-interface.R]: Good catch!

File: tests/testthat/test-dm.R [hunk 3]
```diff
@@ -271,7 +269,6 @@ test_that("`pull_tbl()`-methods work", {
 
   expect_dm_error(
     new_dm3(dm_for_filter %>%
-      dm_zoom_to(t1) %>%
       dm_get_def() %>%
       mutate(zoom = list(t1)), zoomed = TRUE) %>%
```
Comment (krlmlr) [inline tests/testthat/test-dm.R]: How does this work? Do we need instead:

```suggestion
      mutate(zoom = list(t1)), zoomed = TRUE) %>%
      new_dm3() %>%
```

File: tests/testthat/test-learn.R [hunk 4]
```diff
@@ -70,6 +70,7 @@ test_that("Learning from Postgres works?", {
 })
 
 test_that("Learning from SQLite works (#288)?", {
+  skip("learning from SQLite NYI")
```
Comment (krlmlr) [inline tests/testthat/test-learn.R]: Can we backport this fix? I think it's e8133c7.

```suggestion
```

File: tests/testthat/test-zzx-deprecated.R [hunk 5]
```diff
@@ -1,110 +1,95 @@
+withr::local_options(c(lifecycle_verbosity = "quiet"))
```
Comment (krlmlr) [inline tests/testthat/test-zzx-deprecated.R]: It's slightly safer to have them inside every `test_that()` call, these calls shouldn't take too much time. Also, `rlang::local_options()` is slightly faster.

Something else: all of these tests can be skipped on CRAN.

Comment (krlmlr) [reply tests/testthat/test-zzx-deprecated.R]: Scratch that. Let's keep one `local_options()` call at the top of the test file, it's safe enough and easier to read.

File: tests/testthat/test-dm.R (hunk 3 above)
Comment (TSchiefer) [reply tests/testthat/test-dm.R]: The idea is to have multiple tables at once in the `zoom`-column and your suggestion does that, and looks slightly more transparent at the same time.

Comment (TSchiefer) [reply tests/testthat/test-dm.R]: Sorry, misread, `zoomed = TRUE` is an argument for `new_dm3()`. 
I will do it like this:
```r
dm_get_def(dm_for_filter) %>%
      mutate(zoom = list(t1)) %>%
      new_dm3(zoomed = TRUE) %>%
      pull_tbl()
```

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**